src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -611,6 +614,8 @@ protected override void Dispose(bool disposing)
                         base.Dispose(disposing);
                     }
                 }  // finally
+
+                s_byteBufferPool.Free(_buffer);
```

Comment:
It does not, they are all the same size. 32Kb and 8Kb (all below the LOH threshold).